src/index.ts

```diff
@@ -6,28 +6,35 @@ export type { RestEndpointMethodTypes } from "@octokit/plugin-rest-endpoint-meth
 
 import { VERSION } from "./version";
 import type { OctokitOptions } from "@octokit/core/dist-types/types";
-import { HttpsProxyAgent } from "https-proxy-agent";
+import { fetch as undiciFetch, ProxyAgent } from "undici";
 
 const DEFAULTS = {
   authStrategy: createActionAuth,
   baseUrl: getApiBaseUrl(),
   userAgent: `octokit-action.js/${VERSION}`,
 };
 
-function getProxyAgent() {
+export function getProxyAgent() {
```

Comment:
Note that I've exported `getProxyAgent` here and `customFetch` below. There may be a better way around this; I'm not sure.

Reply:
no that's okay. We don't need to document it. I sometimes add a comment stating that this is exported for testing. Another way is to move the function into a separate file if you don't want to add an export to the main file. But in this case I think we are good,

Reply:
Sounds good, thank you for your thoughts!